packages/sync-actions/src/utils/create-build-actions.js

```diff
@@ -1,7 +1,88 @@
+import isEqual from 'lodash.isequal'
+
 function applyOnBeforeDiff(before, now, fn) {
   return fn && typeof fn === 'function' ? fn(before, now) : [before, now]
 }
 
+function isDateOverlap(
+  oldValidFrom,
+  oldValidUntil,
+  newValidFrom,
+  newValidUntil
+) {
+  if (newValidFrom > oldValidFrom && newValidFrom < oldValidUntil) return true
+  if (newValidUntil < oldValidUntil && newValidUntil > oldValidFrom) return true
+  return false
+}
+
+function getPriceId(newPrice, oldVariantArray) {
+  let newPriceId
+
+  const newPriceComparison = {
+    value: { currencyCode: newPrice.value.currencyCode },
+    channel: newPrice.channel,
+    country: newPrice.country,
+    customerGroup: newPrice.customerGroup,
+  }
+
+  oldVariantArray.map(oldVariant =>
```

Comment:
Please do not add type information to variable names. It's not really a good practice. I know that the `sync-actions` use `old` and `new` as wordings. I think `next` and `previous` is actually better. This could mean these would become `previousVariants` or - in case you do decide not to rename - `oldVariants`.